UITests/BookmarksPanelTests.swift

```diff
@@ -62,4 +66,36 @@ class BookmarksPanelTests: KIFTestCase {
         tester().waitForView(withAccessibilityLabel: "https://people-mozilla.org/~npark")
         
     }
+
+    func testRootHasLocalAndBuffer() {
+        let application = UIApplication.shared
+        guard let delegate = application.delegate as? TestAppDelegate else {
+            XCTFail("Couldn't get app delegate.")
+            return
+        }
+
+        let profile = delegate.getProfile(application)
+        guard let bookmarks = profile.bookmarks as? BookmarkBufferStorage else {
+            XCTFail("Couldn't get buffer storage.")
+            return
+        }
+
+        // Add buffer data, then later in the test verify that the buffer mobile folder is not shown in there anymore.
+        createSomeBufferBookmarks()
+
+        let rec1 = BookmarkMirrorItem.bookmark("guid0", modified: Date.now(), hasDupe: false, parentID: BookmarkRoots.MobileFolderGUID, parentName: BookmarksFolderTitleMobile, title: "xyz", description: "xyz", URI: "http://getfirefox.com", tags: "[]", keyword: nil)
+        let rec2 = BookmarkMirrorItem.folder("mobile", modified: Date.now(), hasDupe: false, parentID: "places", parentName: "", title: "Mobile stuff", description: "foo", children: ["guid0"])
```

Comment:
To recap: change the test you copied from, too! Existing flaw. See how TestBookmarkTreeMerging.swift does it.